mico-core/src/main/java/io/github/ust/mico/core/resource/ApplicationResource.java

```diff
@@ -323,15 +325,16 @@
 
         // TODO: Update actual Kubernetes deployment (see issue mico#416).
 
-        return ResponseEntity.ok(new Resource<>(serviceDeploymentInfoDTO, linkTo(methodOn(ApplicationResource.class)
-            .getServiceDeploymentInformation(shortName, version, serviceShortName)).withSelfRel()));
+		return ResponseEntity.ok(new Resource<>(serviceDeploymentInfoResponseDTO,
```

Comment:
`serviceDeploymentInfoResponseDTO` might be `null`. I think that should never happen. Consider to throw an error if this is the case...

Reply:
Actually can **not** be `null` at this point since this is checked in line 305.

Reply:
I meant the `serviceDeploymentInfoResponseDTO` that is created inside the if statement (line 317). If the short name does not match, it will be `null`. However we should throw an error instead.

Reply:
Correct me if i'm wrong, but if the `serviceShortName` does not match any service in the deployment infos, line 306 throws an exception, or am i missing something?

Reply:
Okay you're right. In IntelliJ I have a warning about that, because it does not recognize that the if clause in 305 already checks that.